torchci/pages/api/drci/drci.ts

```diff
@@ -4,11 +4,97 @@ import fetchRecentWorkflows from "lib/fetchRecentWorkflows";
 import { RecentWorkflowsData } from "lib/types";
 import * as drciUtils from "lib/drciUtils";
 
+interface PRandJobs {
+    head_sha: string;
+    pr_number: number;
+    owner_login: string;
+    jobs: RecentWorkflowsData[];
+}
 
 export async function fetchWorkflows() {
     const recentWorkflows: RecentWorkflowsData[] = await fetchRecentWorkflows(
         drciUtils.NUM_MINUTES + ""
     );
+
+    const workflowsByPR = reorganizeWorkflows(recentWorkflows);
+
+    for (const [pr_number, pr] of workflowsByPR) {
+        const { pending, failedJobs } = getWorkflowAnalysis(pr);
+        const failureInfo = constructFailureAnalysis(pending, failedJobs, pr.head_sha);
+        //add failureInfo to Dr. CI comment on each PR
+    }
+}
+
+export function constructFailureAnalysis(
+    pending: number,
+    failedJobs: RecentWorkflowsData[],
+    sha: string
+): string {
+    let output = ``;
+    const failing = failedJobs.length;
+    const noneFailing = `## :white_check_mark: No Failures`;
+    const someFailing = `## :x: ${failing} Failures`;
+    const somePending = `, ${pending} Pending`;
+    if (failing === 0) {
```

Comment:
nit you can extract this out to be like
```
hasPending = pending > 0;
hasFailing = failing > 0;
```
and this would get rid of alot of the comparison checks and make it a bit more clear.